src/explore-education-statistics-frontend/src/modules/data-catalogue/DataCataloguePage.tsx

```diff
@@ -63,6 +67,13 @@ export const getServerSideProps: GetServerSideProps<Props> = async context => {
         publicationFilter: 'DataCatalogue',
       }),
     );
+
+    if (!showTypeFilter) {
+      const apiDataSets = await queryClient.fetchQuery(
+        dataSetFileQueries.list({ dataSetType: 'api' }),
+      );
+      showTypeFilter = !!apiDataSets.results.length;
```

Comment:
Not seen the double `!!` before, does it behave similarly to `!apiDataSets.results`?

Reply:
It's a shorthand way of doing something like `showTypeFilter = apiDataSets.results.length ? true : false;`. It's essentially using the not operator (`!`) twice to make an inverted boolean. We need to check for the length of array as just checking results exists would return true for an empty results array.